lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -282,97 +408,156 @@ def parse_genbank(self, file_path, params):
-            if not re.match("\d+\/\d+\/\d+", assembly_ref):
-                raise ValueError(f"Assembly ref: {assembly_ref} is not a valid format. Must"
-                                 f" be in numerical <ws>/<object>/<version> format.")
-            ret = self.dfu.get_objects(
-                {'object_refs': [assembly_ref]}
-            )['data'][0]
-            if "KBaseGenomeAnnotations.Assembly" not in ret['info'][2]:
-                raise ValueError(f"{assembly_ref} is not a reference to an assembly")
-            unmatched_ids = list()
-            unmatched_ids_md5s = list()
-            for current_contig in self.contig_seq.keys():
-                current_contig_md5 = hashlib.md5(
-                    str(self.contig_seq[current_contig]).encode('utf8')
-                ).hexdigest()
-                if current_contig in ret['data']['contigs']:
-                    if current_contig_md5 != ret['data']['contigs'][current_contig]['md5']:
-                        unmatched_ids_md5s.append(current_contig)
-                else:
-                    unmatched_ids.append(current_contig)
-            if len(unmatched_ids) > 0:
-                raise ValueError(warnings['assembly_ref_extra_contigs'].format(", ".join(unmatched_ids)))
-            if len(unmatched_ids_md5s) > 0:
-                raise ValueError(warnings["assembly_ref_diff_seq"].format(", ".join(unmatched_ids_md5s)))
-            logging.info(f"Using supplied assembly: {assembly_ref}")
-            return assembly_ref
-        logging.info("Saving sequence as Assembly object")
-        Bio.SeqIO.write(out_contigs, fasta_file, "fasta")
-        assembly_ref = self.aUtil.save_assembly_from_fasta(
-            {'file': {'path': fasta_file},
-             'workspace_name': params['workspace_name'],
-             'assembly_name': assembly_id,
-             'type': params.get('genome_type', 'isolate'),
-             'contig_info': extra_info})
-        logging.info(f"Assembly saved to {assembly_ref}")
-        return assembly_ref
+            genome_obj.contig_seq[in_contig.id] = in_contig.seq.upper()
+        return out_contigs, extra_info
+
+    def _save_assemblies(
+        self,
+        workspace_id,
+        genbank_files,
+        inputs,
+        genome_objs,
+    ):
+        name2ref = {}
+        assembly_ids = []
+        fasta_files = []
+        extra_infos = []
+        types = []
+
+        for genbank_file, params, genome_obj in zip(
+            genbank_files, inputs, genome_objs
+        ):
+            contigs = Bio.SeqIO.parse(genbank_file, "genbank")
+            assembly_id = f"{params['genome_name']}_assembly"
+            fasta_file = f"{self.cfg.sharedFolder}/{assembly_id}.fasta"
+
+            out_contigs, extra_info = self._get_contigs_and_extra_info(
+                contigs, genome_obj
+            )
+
+            assembly_ref = params.get("use_existing_assembly")
+            if assembly_ref:
+                self._validate_existing_assembly(
```

Comment:
This is another case where we're fetching assemblies one at a time, but I don't think it's used very often and so probably we can not worry about it for now
https://github.com/search?q=org%3Akbaseapps+use_existing_assembly+-path%3AGenomeFileUtil&type=code

Reply:
👍

Reply:
:+1: